### ISSUE TITLE
Bug: no-unused-vars shows only for 1 if result

### ISSUE THREAD

**Issue by gadzbi123** [eslint/eslint#16049]
### Environment

Node version: v16.15.0
npm version: v8.5.5
Local ESLint version: v8.18.0 (Currently used)
Global ESLint version: v8.18.0
Operating System: win32 10.0.22000

### What parser are you using?

Other

### What did you do?

<details>
<summary>Configuration</summary>

```
```
</details>

```js
const time1 = 25;
let string1;
time1 > 10 && time1 < 20 ? (string1 = 'good') : (string1 = 'bad');

```

### What did you expect to happen?

Should't it give me warning(error) for the first string assignment?
Or at least give me the warning(error) for both results of if statement.

### What actually happened?

It have gave me the warning (error) just for the second string assignment. 

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I think it's not a huge issue but it could be done better.

**Comment by mdjermanovic** [comment]
Hi @gadzbi123, thanks for the issue!

This works as intended. This rule reports unused variables (those that are never read), but not every unused assignment to the variable.

For example, the following code is valid for this rule:

```js
/* eslint no-unused-vars: "error" */

let string1;

if (foo) {
  string1 = "good"; // this assigned value is unused, but it isn't reported
}

string1 = "bad";

bar(string1); // the variable is read here, so it is used

```

[ESLint Playground link](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVudXNlZC12YXJzOiBcImVycm9yXCIgKi9cblxubGV0IHN0cmluZzE7XG5cbmlmIChmb28pIHtcbiAgc3RyaW5nMSA9IFwiZ29vZFwiOyAvLyB0aGlzIGFzc2lnbmVkIHZhbHVlIGlzIHVudXNlZCwgYnV0IGl0IGlzbid0IHJlcG9ydGVkXG59XG5cbnN0cmluZzEgPSBcImJhZFwiO1xuXG5iYXIoc3RyaW5nMSk7IC8vIHRoZSB2YXJpYWJsZSBpcyByZWFkIGhlcmUsIHNvIGl0IGlzIHVzZWQiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==)

By design, if the variable has write references, the last one will be reported instead of the declaration, and that's why in your example it reports `string1 = 'bad'`.
